engine/function/src/templates/Numeric.ftl

```diff
@@ -1226,13 +1267,38 @@ public class Numeric {
         int n = values.intSize("median");
 
         if (n == 0) {
-            return Double.NaN;
+            return NULL_DOUBLE;
         } else {
-            ${pt.primitive}[] copy = values.copyToArray();
-            Arrays.sort(copy);
-            if (n % 2 == 0)
-                return 0.5 * (copy[n / 2 - 1] + copy[n / 2]);
-            else return copy[n / 2];
+            // NULL values sorted to beginning of the array.
+            ${pt.primitive}[] copy = sort(values.toArray());
```

Comment:
This is not ideal. We should be embarrassed by the implementation of `sort`:
1. We make one `toArray()` call internally.
2. Then we make an array of boxed values.
3. Then we sort using a comparator.
4. Then we make a third copy wherein we *unbox* the values.

Sorting on the vector at least let's us skip *one* of the copies.

We should write a separate ticket for updating the sort kernels in `io.deephaven.engine.table.impl.sort` to have a version that does not bother with a `valuesToPermute` chunk, and use that version for the implementation of `sort`. This way we make exactly one `copyToArray()` and we're done. Note `copyToArray()`, not `toArray()`: we mustn't sort our backing array.